pkgs/applications/video/obs-studio/wrapper.nix

```diff
@@ -8,11 +8,17 @@ symlinkJoin {
   nativeBuildInputs = [ makeWrapper ];
   paths = [ obs-studio ] ++ plugins;
 
-  postBuild = ''
-    wrapProgram $out/bin/obs \
-      --set OBS_PLUGINS_PATH      "$out/lib/obs-plugins" \
-      --set OBS_PLUGINS_DATA_PATH "$out/share/obs/obs-plugins"
-  '';
+  postBuild =
+    let
+      # Some plugins needs extra environment, see obs-gstreamer for an example.
+      pluginArguments =
+        lib.lists.foldr (item: accu: (item.passthru.obsWrapperArguments or []) ++ accu) [] plugins;
```

Comment:
I am not sure if you really need this `foldr` here. Doesn't something like `flatten` and just simply list concatenation wouldn't work here?

Reply:
Sure, I'm so used to simple std libraries that I forgot nixpkgs could have something easier. I do need a map to do that `or []`. But I'll push changes....